django/contrib/postgres/fields/serial.py

```diff
@@ -0,0 +1,77 @@
+from django.db import models
+from django.db.models.expressions import DatabaseDefault, Expression
+from django.utils.translation import gettext_lazy as _
+
+__all__ = ("BigSerialField", "SmallSerialField", "SerialField")
+
+
+class SerialFieldSequence(Expression):
+    allowed_default = True
+
+    def as_sql(self, compiler, connection):
+        table = self.field.model._meta.db_table
+        column = self.field.column
+        return f"nextval(pg_get_serial_sequence('{table}', '{column}'))", []
+
+
+class SerialFieldMixin:
+    db_returning = True
+
+    def __init__(self, *args, **kwargs):
+        default = DatabaseDefault()
+        db_default = SerialFieldSequence(self)
+
+        if not kwargs.setdefault("blank", True):
+            raise ValueError(f"{self.__class__.__name__} must be blank.")
+        if kwargs.setdefault("null", False):
+            raise ValueError(f"{self.__class__.__name__} must not be null.")
+        if kwargs.setdefault("default", default) is not default:
+            raise ValueError(f"{self.__class__.__name__} cannot have a default.")
+        if kwargs.setdefault("db_default", db_default) is not db_default:
+            raise ValueError(
+                f"{self.__class__.__name__} cannot have a database default."
+            )
+
+        super().__init__(*args, **kwargs)
+
+    def deconstruct(self):
+        name, path, args, kwargs = super().deconstruct()
+        del kwargs["default"]
+        del kwargs["db_default"]
+        return name, path, args, kwargs
+
+    def get_prep_value(self, value):
+        value = super().get_prep_value(value)
+        if value is None:
+            value = self.get_default()
+        return value
```

Comment:
I'm surprised this is required given we set `default = DatabaseDefault()`?

Reply:
It's for supporting `None` arguments:

`SerialModel.objects.create(serial=None)`

Since it's possible to provide `None` to auto fields, it should also be possible to provide it for serial fields.

If that's not necessary, we can remove it.